lib/topological_inventory/amazon/collector.rb

```diff
@@ -177,6 +177,36 @@ def default_region
       def inventory_name
         "Amazon"
       end
+
+      def paginated_query(scope, connection, collection_name)
+        func = lambda do |&blk|
+          send(connection, scope).client.public_send("describe_#{collection_name.to_s}").each do |result|
+            result.public_send(collection_name).each do |item|
+              blk.call(item, scope)
+            end
+          end
+        end
+        Iterator.new(func, "Couldn't fetch '#{collection_name}' from #{connection} with #{scope}")
+      end
+
+      def manual_paginated_query(scope, connection, collection_name)
```

Comment:
@Ladas where is this used?

Reply:
it's not, I just wanted to make sure the iterator above worked fine

Reply:
ok removed